flask_project/campaign_manager/test/test_models.py

```diff
@@ -167,169 +134,120 @@ def test_04_geom_campaign(self):
             asShape(geom),
             srid=4326
             )
-        mock_campaign = session.query(Campaign).first()
+        mock_campaign = campaign.get_by_uuid("mock_campaign_uuid")
         mock_taskBoundary = TaskBoundary(
             coordinates=geom_obj,
             name="name",
             status="status",
             type_boundary="type_boundary",
             campaign_id=mock_campaign.id
             )
-        session.commit()
-        self.assertEqual(mock_taskBoundary.coordinates, geom_obj)
+        mock_taskBoundary.create()
+        coor_geoJSON = mock_campaign.get_task_boundary_as_geoJSON()
+        geom_check = (geom == ast.literal_eval(coor_geoJSON[0]))
+        self.assertTrue(geom_check)
 
     def test_05_default_feild_condition(self):
-        mock_campaign = session.query(Campaign).first()
+        campaign = Campaign()
+        mock_campaign = campaign.get_by_uuid("mock_campaign_uuid")
         self.assertFalse(mock_campaign.link_to_OpenMapKit)
 
     def test_06_delete_campaign(self):
```

Comment:
campaign.delete() should delete task_boundary and user.
why are you deleting the user?

Reply:
Nosetest are called in a random order, any class can be called first, so I try to clean the database after every class. that is the reason I am deleting user or else I might get a duplicate key error.
The correct way is to teardown and setup DB after every test.
I am doing this right now but after the tests are run, nosetest gets stuck.